systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaTopicUtils.java

```diff
@@ -167,6 +168,35 @@ public static boolean waitForKafkaTopicNotReady(String topicName) {
         return waitForKafkaTopicStatus(kubeClient().getNamespace(), topicName, NotReady);
     }
 
+    public static void waitForTopicConfigContains(String namespaceName, String topicName, Map<String, Object> config) {
+        LOGGER.info("Wait until topic {} contains correct config", topicName);
+        TestUtils.waitFor("Wait for correct config",
+                Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_STATUS_TIMEOUT,
+                () -> KafkaTopicUtils.configsAreEqual(KafkaTopicResource.kafkaTopicClient()
+                        .inNamespace(namespaceName).withName(topicName).get().getSpec().getConfig(), config)
+        );
+        LOGGER.info("Topic {} contains correct config", topicName);
+    }
+
+    public static boolean configsAreEqual(Map<String, Object> actualConf, Map<String, Object> expectedConf) {
+        if (expectedConf == null) {
+            LOGGER.debug("Expected config is null");
+            return false;
+        }
+        if (actualConf == null) {
+            LOGGER.debug("Actual config is null");
+            return false;
+        }
+
+        if (expectedConf.size() != actualConf.size()) {
+            LOGGER.debug("Configs are not the same length");
+            return false;
+        }
+
+        return expectedConf.entrySet().stream()
+                    .allMatch(expected -> expected.getValue().toString().equals(actualConf.get(expected.getKey()).toString()));
```

Comment:
I would maybe simplify this a little bit. You don't have to have those 3 returns.
What about:
```suggestion
        if (expectedConf != null && actualConf != null) {
            return expectedConf.entrySet().stream()
                    .allMatch(expected -> expected.getValue().toString().equals(actualConf.get(expected.getKey()).toString()));
        }
        
        return false;
```